### PR TITLE
Fixes 20220309

### PR DESCRIPTION
**Pull request by eoliveros**

### CODE REVIEW

File: app_core.py [hunk 1]
```diff
@@ -9,8 +9,27 @@
 from addresswatcher import AddressWatcher
 from timer import Timer
 
+class ReverseProxied(object):
+    """
+    Because we are reverse proxied from an aws load balancer
```
Comment (djpnewton) [inline app_core.py]:
```suggestion
    Because we are reverse proxied from an http/s server
```

File: app_core.py [hunk 2]
```diff
@@ -9,8 +9,27 @@
...
+    use environ/config to signal https
+    since flask ignores preferred_url_scheme in url_for calls
+    """
+
+    def __init__(self, app):
+        self.app = app
+
+    def __call__(self, environ, start_response):
+        # if one of x_forwarded or preferred_url is https, prefer it.
+        forwarded_scheme = environ.get("HTTP_X_FORWARDED_PROTO", None)
```
Comment (djpnewton) [inline app_core.py:24]: does it crash is HTTP_X_FORWARDED_PROTO or PREFERRED_URL_SCHEME are not present in the envrionment?

**Review by djpnewton — CHANGES_REQUESTED** [review]

File: app_core.py (hunk 1 above)
Comment (eoliveros) [reply app_core.py]: changed to what was suggested above.

File: app_core.py (hunk 2 above)
Comment (eoliveros) [reply app_core.py:24]: changed the variable to PREFERRED_URL_SCHEME_2 and it seems to work.

**Pull request merged**